packages/share/android/src/main/java/io/flutter/plugins/share/SharePlugin.java

```diff
@@ -36,6 +37,14 @@ public void onMethodCall(MethodCall call, MethodChannel.Result result) {
       // Android does not support showing the share sheet at a particular point on screen.
       share((String) call.argument("text"));
       result.success(null);
+
+    } else if (call.method.equals("shareFile")) {
+      if (!(call.arguments instanceof Map)) {
+        throw new IllegalArgumentException("Map argument expected");
+      }
```

Comment:
A suitable runtime exception (`ClassCastException`) is [already being thrown](https://github.com/flutter/engine/blob/master/shell/platform/android/io/flutter/plugin/common/MethodCall.java#L72) by the `call.argument(String)` method, if the arguments are not as expected. Sending a map with non-String values would also result in a `ClassCastException` in the proposed code, leaving us with two different exception types for the same kind of programming error.

The `MethodCall` API was designed to avoid the need for explicit type error checking and casting in plugin implementations. You can basically declare your type expectations with code like the following:
```java
String uri = call.argument("uri");
String mimeType = call.argument("mimeType");
shareFile(uri, mimeType);
```